### PR TITLE
Support MaximumNumberOfIterations = 0, allowing to just do an AutomaticTransformInitialization

### PR DESCRIPTION
**Pull request by N-Dekker**
Until now, optimizers appear to do a single iteration step, even when the elastix parameter "MaximumNumberOfIterations" has the value "0". It appears that those optimizers did the check `if (m_CurrentIteration >= m_NumberOfIterations) break;` when it was too late already to avoid the first iteration. 

Supporting MaximumNumberOfIterations = 0 (by doing no optimizer iteration at all) allows inspecting the result of just doing an AutomaticTransformInitialization. As hereby tested.

### CODE REVIEW

**Comment by mstaring** [comment]
lgtm

**Comment by N-Dekker** [comment]
Thanks @mstaring  Just wondering, are there already other ways to only just run the AutomaticTransformInitialization (from the elastix.exe command-line or the Elastix Filter interface)? Or is this a new feature?  😃

**Comment by mstaring** [comment]
Let's actually discuss at the sprint, because now I am unsure if this is what we want. 
If I recall correctly with just 0 iteration the value and derivative are compute but never applied.

**Comment by N-Dekker** [comment]
@mstaring  It appears to me that until now MaximumNumberOfIterations = 0 was treated exactly like MaximumNumberOfIterations = 1. 

Whatever we do, I think we should not treat 0 and 1 as equal!

File: Core/Main/GTesting/itkElastixRegistrationMethodGTest.cxx [hunk 1]
```diff
@@ -87,6 +87,103 @@ GTEST_TEST(itkElastixRegistrationMethod, Translation)
...
+                                                       { "Metric", "AdvancedNormalizedCorrelation" },
+                                                       { "Optimizer", optimizer },
+                                                       { "Transform", "TranslationTransform" } }));
+    filter->Update();
+
+    const auto transformParameters = GetTransformParametersFromFilter(*filter);
+
+    for (const auto & transformParameter : transformParameters)
+    {
+      EXPECT_EQ(transformParameter, 0.0);
```
Comment (N-Dekker) [inline Core/Main/GTesting/itkElastixRegistrationMethodGTest.cxx:133]: During the sprint, Stefan (@stefanklein) suggested to add an extra test that the transformParameters are _not_ all zero when MaximumNumberOfIterations = 1. This appears to take some more time, as in the current setup, it would cause exceptions for "FiniteDifferenceGradientDescent" and "StandardGradientDescent".  (MaximumNumberOfIterations = 1 only works out of the box with "AdaptiveStochasticGradientDescent", in this specific test case.) So I would like to postpone this extra test for later.... 😃
